script/constructron.lua

```diff
@@ -799,46 +727,76 @@ me.conditions = {
+            if job.path_active and ((game.tick - job.start_tick) > 120) then
+                if (chunk_util.distance_between(constructron.position, position) > 5) then -- the condition
+                    if not constructron.autopilot_destination then -- path lost recovery
+                        job.start_tick = game.tick
+                        me.actions[job.action](job, table.unpack(job.action_args or {}))
+                    elseif constructron.speed < 0.2 and ((game.tick - job.start_tick) > 600) then -- stuck recovery
+                        if ((job.attempt or 1) < 4) or job.returning_home then
+                            if job.landfill_job then -- is this a landfill job?
+                                if not constructron.logistic_cell.logistic_network.can_satisfy_request("landfill", 1) then
+                                    -- !! Logic gap - Constructrons will return home even if there is other entities to build.
+                                    me.graceful_wrapup(job) -- no landfill left.. leave
+                                    return
+                                end
+                            end
+                            local mvmt_last_distance = job.mvmt_last_distance
+                            local distance
+                            distance = chunk_util.distance_between(constructron.position, constructron.autopilot_destination)
+                            if mvmt_last_distance and (mvmt_last_distance - distance < 2) and constructron.speed < 0.1 then -- check that movement has progressed at least two tiles in the last 10 seconds.
+                                job.mvmt_last_distance = nil
+                                job.start_tick = game.tick
+                                me.actions[job.action](job, table.unpack(job.action_args or {}))
+                            else
+                                job.mvmt_last_distance = distance
+                                job.start_tick = game.tick
+                            end
+                        else
+                            me.graceful_wrapup(job) -- unable to reach destination
+                            return
+                        end
+                    end
                     return false  -- condition is not met
-                else
-                    return "graceful_wrapup" -- unable to reach destination
                 end
+            else
+                -- game.print('path not yet started') --testing
+                return false
             end
         else
             return true  -- leader is invalidated.. skip action
         end
-        me.enable_roboports(constructrons[1].grid)
+        -- possibly should check that the Constructron is stopped
+        me.enable_roboports(constructron.grid)
         return true -- condition is satisfied
     end,
 
     ---@param job Job
     ---@param _ any
     ---@return boolean | string
     build_done = function(job, _, _, _)

```

Comment:
the return value annotation seems to be incorrect now.
```suggestion
    ---@param job Job
    ---@param _ any
    ---@return boolean?
    build_done = function(job, _, _, _)
```

Reply:
Fixed